apps/tasks.py

```diff
@@ -649,3 +490,161 @@ def delete_old_objects():
+                ):
+                    continue
+
+                # If pod is same and deleted, set deleted stamp
+                elif creation_timestamp == status_data[release]["creation_timestamp"] and deletion_timestamp:
+                    status = "Deleted"
+                    appinstance.deleted_on = timezone.now()
+
+                # If pod is newer, update
+                status_data[release] = {
+                    "creation_timestamp": creation_timestamp,
+                    "deletion_timestamp": deletion_timestamp,
+                    "status": status,
+                }
+                status_object = AppStatus(appinstance=appinstance)
+                update_status(appinstance, status_object, status)
+    except Exception as exc:
+        # Catch other exceptions to trigger a retry
+        raise self.retry(exc=exc, countdown=10)
+
+
+@worker_ready.connect
+def on_worker_ready(**kwargs):
+    """
+    This function starts the event listener in a given namespace.
+    When the Celery worker is ready, the task is started.
+    """
+    label_selector = "type=app"
+    NAMESPACE = settings.NAMESPACE
+    sync_all_statuses(namespace=NAMESPACE, label_selector=label_selector)
+    init_event_listener.apply_async(args=(NAMESPACE, label_selector), countdown=1)
+
+
+def setup_client():
+    """
+    Sets up the kubernetes python client and event streamer
+    """
+    try:
+        config.load_incluster_config()
+    except config.ConfigException:
+        try:
+            config.load_kube_config(settings.KUBECONFIG)
+        except config.ConfigException as e:
+            raise config.ConfigException(
+                "Could not set the cluster config. Try to use the cluster.conf file or set incluster config"
+            ) from e
+
+    k8s_api = client.CoreV1Api()
+
+    return k8s_api
+
+
+def get_status(pod):
+    """
+    Returns the status of a pod by looping through each container
+    and checking the status.
+    """
+    container_statuses = pod.status.container_statuses
+
+    if container_statuses is not None:
+        for container_status in container_statuses:
+            state = container_status.state
+
+            if state is not None:
+                terminated = state.terminated
+
+                if terminated is not None:
+                    reason = terminated.reason
+                    return mapped_status(reason)
+
+                waiting = state.waiting
+
+                if waiting is not None:
+                    reason = waiting.reason
+                    return mapped_status(reason)
+        else:
+            running = state.running
+            ready = container_status.ready
+            if running and ready:
+                return "Running"
+            else:
+                return "Pending"
+
+    return pod.status.phase
+
+
+K8S_STATUS_MAP = {
+    "CrashLoopBackOff": "Error",
+    "Completed": "Retrying...",
+    "ContainerCreating": "Created",
+    "PodInitializing": "Pending",
+}
```

Comment:
I suggest you move it to after imports :)

Reply:
Fixed!